### PR TITLE
Derive some useful traits

### PR DESCRIPTION
**Pull request by orzogc**

### CODE REVIEW

File: pretend/src/lib.rs [hunk 1]
```diff
@@ -334,6 +335,7 @@ impl<T> Response<T> {
 /// be implemented by this struct.
 ///
 /// See crate level documentation for more information
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
```
Comment (SfietKonstantin) [inline pretend/src/lib.rs]: I'm not sure that you want to compare a client, so Eq, PartialEq and Hash might not make sense too much ...

File: pretend/src/resolver.rs [hunk 2]
```diff
@@ -24,6 +24,7 @@ pub trait ResolveUrl {
 /// Default URL resolver
 ///
 /// This resolver appends the path to a base URL.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
```
Comment (SfietKonstantin) [inline pretend/src/resolver.rs]: Same as for Pretend, this is a rather technical struct, so Clone, Debug make sense, but not really Eq, Ord and Hash.

**Review by SfietKonstantin — CHANGES_REQUESTED** [review]
You might also want to implement Clone, Debug for pretend-awc.

Otherwise, thanks for these useful contributions :)

**Comment by orzogc** [comment]
> You might also want to implement Clone, Debug for pretend-awc.
> 
> Otherwise, thanks for these useful contributions :)

[awc::Client](https://docs.rs/awc/2.0.3/awc/struct.Client.html) doesn't implement the `Debug` trait.

**Comment by SfietKonstantin** [comment]
Thanks for this contribution !
I will merge (and have a look at awc)

**Pull request merged**